### PR TITLE
HPCC-13590 Fix IF() row accessing row that has been destroyed

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@richardkchapman the quick fix for the if() row problem was simpler than I expected.
@jamienoss please review

File: roxie/roxiemem/roxiemem.cpp [hunk 1]
```diff
@@ -27,6 +27,7 @@
 
 #ifdef _DEBUG
 #define _CLEAR_ALLOCATED_ROW
+//#define _CLEAR_FREED_ROW
```
Comment (ghalliday) [inline roxie/roxiemem/roxiemem.cpp]: I would be tempted to always enable this in debug mode.  What do you think?

Comment (richardkchapman) [inline roxie/roxiemem/roxiemem.cpp]: So long as it does not distort timings too much

**Comment by AttilaVamos** [comment]
Automated Smoketest Build: success
HPCC Start: OK

| setup | hthor | thor | roxie |
| ----- | ----- | ----- | ---- |
| setup | Pass | Pass | Pass |
| setup_fetch | Pass | Pass | Pass |
| setupdict | Pass | Pass | Pass |
| setupsearchindex | Pass | Pass | Pass |
| setupsq | Pass | Pass | Pass |
| setupwordindex | Pass | Pass | Pass |
| setupxml | Pass | Pass | Pass |
 

| test | hthor | thor | roxie |
| ----- | ----- | ----- | ---- |
| issue13590 | Pass | Pass | Pass |
 
HPCC Stop: OK

**Comment by jamienoss** [comment]
@richardkchapman. @ghalliday this looks  good to me other than the decision of a debug default for _CLEAR_FREED_ROW

**Pull request merged**